src/common/libsubprocess/server.c

```diff
@@ -241,9 +241,11 @@ static int rexec_output (flux_subprocess_t *p,
                          bool eof)
 {
     json_t *io = NULL;
+    char rankstr[64];
     int rv = -1;
 
-    if (!(io = ioencode (stream, s->rank, data, len, eof))) {
+    snprintf (rankstr, sizeof (rankstr), "%d", s->rank);
+    if (!(io = ioencode (stream, rankstr, data, len, eof))) {
```

Comment:
Since this is a recurring theme, would it make sense to have an ioencode interface for it like `ioencode_rank()` that takes an integer rank like before?

Reply:
I mentioned this in #2437.  I was going to make a `ioencode_rank()` until I realized the above pattern was only used twice.  So IMO it didn't quite meet the "convenience function" threshold.  Are you thinking we should add for future under assumption of increased use.

Reply:
Ah now that you mention it I saw that.  No, I'm fine with that decision if it's not likely to be a recurring need, sorry for the noise.